src/layers.jl

```diff
@@ -0,0 +1,175 @@
+"""
+PUSH_TO_LAYER
+
+A setinel to aid the creation of layers.
+If set to true, all the objects are pushed to the current layer.
+Can be accessed using PUSH_TO_LAYER[1]
+"""
+const PUSH_TO_LAYER = Array{Any,1}()
+# by default push all the objects to the video
+push!(PUSH_TO_LAYER, false)
+
+"""
+    @JLayer(frames, width, height, position, body)
+Calls the [`to_layer_m`](@ref) method to create a [`Layer`](@ref) out of the arguments given. 
+
+# Arguments
+- `frames`:a `UnitRange` that defines for which frames the layer is active
+- `width`: defines the width of the layer
+- `height`: defines the height of the layer
+- `position`: location of the center of the layer on the main canvas
+- `transparent` : Whether the layer should have a transparent background(:transparent or :opaque)
+- `body`
+    - It contains all the objects(and thier respective actions) definitions for a layer
+    - A layer can have it's own separate background
+    - Anything defined within the `begin...end` block stays in the layer
+    - A Layer has it's own coordinate reference sysstem, i.e. it has it's own origin 
+    So eg : `Point(100, 100)` is different when defined in a layer and doesn't represent
+    the location 100, 100 on the main canvas 
+
+`width`, `height`, `position` and `transparent` are optional and default to the video's width, height, origin and :opaque respectively.
+Layer declaration should take place before pushing objects to it if one is not using the macro
+
+# Example
+```julia
+function ground(args...)
+    background("white")
+    sethue("black")
+end
+
+video = Video(500, 500)
+Background(1:100, ground)
+object((args...)->circle(O, 50, :fill))
+
+l1 = @Javis.Layer 10:70 100 100 Point(150, 150) :transparent begin
+    red_ball = Object(20:60, (args...)->object(O, "red"), Point(50,0))
+    act!(red_ball, Action(anim_rotate_around(2π, O)))
+end
+render(video; pathname="test.gif")
+```
+"""
+macro JLayer(
+    frames::Expr,
+    width::Int,
+    height::Int,
```

Comment:
These `::Int` are quite counterproductive as the width and height can't be specified as a variable anymore then.

Reply:
The problem with not having them as Int is then the macro doesn't read them as variables holing a value but literally `width` and `height`. This will also create issues with the macro multiple dispatch

Reply:
Yeah I see. Is it possible to have a `@JLayer frames point begin` then ? Otherwise changing the width and height of a video becomes a hassle as one might need to change all the layers when they should be the same size as the background.